transport/ice/IceSession.cpp

```diff
@@ -472,11 +493,252 @@ bool IceSession::isResponseAuthentic(const void* data, size_t len) const
+        return false;
+    }
+
+    IceCandidate newRemoteCandidate(_component,
+        localEndpoint->getTransportType(),
+        remoteCandidatePriority,
+        remotePort,
+        remotePort,
+        IceCandidate::Type::PRFLX);
+
+    newRemoteCandidate = addRemoteCandidate(newRemoteCandidate);
+
+    for (auto& endpoint : _endpoints)
+    {
+        auto* candidatePair = addProbeForRemoteCandidate(endpoint, newRemoteCandidate);
+        if (candidatePair)
+        {
+            candidatePair->nextTransmission = now;
+            candidatePair->receptionTimestamp = now;
+
+            if (endpoint.endpoint == localEndpoint)
+            {
+                candidatePair->accept();
+            }
+        }
+    }
+
+    sendResponse(localEndpoint, remotePort, 0, msg, now);
+    sortCheckList();
+    processTimeout(now);
+    return true;
+}
+
+void IceSession::processValidStunRequest(IceEndpoint* localEndpoint,
+    const transport::SocketAddress& remotePort,
+    const StunMessage& msg,
+    uint64_t now)
+{
+    logger::debug("%s probe from %s -> %s",
+        _logId.c_str(),
+        ice::toString(localEndpoint->getTransportType()).c_str(),
+        remotePort.toString().c_str(),
+        localEndpoint->getLocalPort().toString().c_str());
+
+    int remoteCandidatePriority = 0;
+    const auto prioAttribute = msg.getAttribute<StunPriority>(StunAttribute::PRIORITY);
+    if (prioAttribute)
+    {
+        remoteCandidatePriority = prioAttribute->value;
+    }
+
+    bool successfullyProcessed = false;
+
+    if (localEndpoint->getTransportType() == TransportType::TCP)
+    {
+        successfullyProcessed =
+            processValidStunTcpRequest(localEndpoint, remotePort, msg, remoteCandidatePriority, now);
+    }
+    else if (localEndpoint->getTransportType() == TransportType::UDP)
+    {
+        successfullyProcessed =
+            processValidStunUdpRequest(localEndpoint, remotePort, msg, remoteCandidatePriority, now);
+    }
+
+    if (successfullyProcessed && _credentials.role == IceRole::CONTROLLED)
+    {
+        const auto useCandidate = msg.getAttribute(StunAttribute::USE_CANDIDATE);
+        auto useCandidatePair = useCandidate ? findCandidatePair(localEndpoint, remotePort) : nullptr;
+        if (useCandidatePair)
+        {
+            if (_nomination != useCandidatePair)
+            {
+                logger::debug("remote nominated %s-%s",
+                    _logId.c_str(),
+                    useCandidatePair->localCandidate.address.toString().c_str(),
+                    useCandidatePair->remoteCandidate.address.toString().c_str());
+                if (useCandidatePair->state == ProbeState::Succeeded)
+                {
+                    if (_nomination)
+                    {
+                        // Aggressive nomination
+                        // Then we will clean the old nominated flag
+                        _nomination->nominated = false;
```

Comment:
Add support to agressive nomination when we are CONTROLLED. The code before it was already accepting new nominations but was not clearing the flag